src/Features/Core/Portable/CodeRefactorings/IRefactoringHelpersService.cs

```diff
@@ -34,5 +35,13 @@ internal interface IRefactoringHelpersService : ILanguageService
         /// </para>
         /// </summary>
         Task<TSyntaxNode> TryGetSelectedNodeAsync<TSyntaxNode>(Document document, TextSpan selection, CancellationToken cancellationToken) where TSyntaxNode : SyntaxNode;
+        Task<TSyntaxNode> TryGetSelectedNodeAsync<TSyntaxNode>(Document document, TextSpan selection, Func<TSyntaxNode, bool> predicate, CancellationToken cancellationToken) where TSyntaxNode : SyntaxNode;
+
+        Task<TSyntaxNode> TryGetDeepInNodeAsync<TSyntaxNode>(
```

Comment:
1. this def needs docs.  
2. super bummed we can't keep the simple semantics of the original method.  would very much like to know why this method and the one above it are necessary.